SpatialGDK/Source/SpatialGDK/Private/Tests/WorkerViewTest.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Tests/TestDefinitions.h"
+
+#include "WorkerView.h"
+#include "OpList/ViewDeltaLegacyOpList.h"
+
+#define WORKERVIEW_TEST(TestName) \
+	GDK_TEST(Core, WorkerView, TestName)
+
+using namespace SpatialGDK; 
+
+namespace
+{
+	Worker_Op CreateEmptyCreateEntityResponseOp()
+	{
+		Worker_Op Op{};
+		Op.op_type = WORKER_OP_TYPE_CREATE_ENTITY_RESPONSE;
+		Op.op.create_entity_response = Worker_CreateEntityResponseOp{};
+		return Op;
+	}
+
+} // anonymous namespace
+
+WORKERVIEW_TEST(GIVEN_empty_WorkerView_WHEN_GenerateViewDelta_called_THEN_ViewDelta_returned)
+{
+	WorkerView View;
+	const ViewDelta* Delta = View.GenerateViewDelta();
+
+	TestTrue("Valid ViewDelta returned", Delta != nullptr);
+	return true;
+}
+
+WORKERVIEW_TEST(GIVEN_WorkerView_with_one_CreateEntityRequest_WHEN_FlushLocalChanges_called_THEN_one_CreateEntityRequest_returned)
+{
+	// GIVEN
+	WorkerView View;
+	CreateEntityRequest Request;
+	View.SendCreateEntityRequest(Request);
+
+	// WHEN
+	auto Messages = View.FlushLocalChanges();
+
+	// THEN
+	TestTrue("WorkerView has one CreateEntityRequest", Messages->CreateEntityRequests.Num() == 1);
+
+	return true;
+}
+
+WORKERVIEW_TEST(GIVEN_WorkerView_with_multiple_CreateEntityRequest_WHEN_FlushLocalChanges_called_THEN_mutliple_CreateEntityRequests_returned)
+{
+	// GIVEN
+	WorkerView View;
+	CreateEntityRequest Request;
+	View.SendCreateEntityRequest(Request);
+	View.SendCreateEntityRequest(Request);
+
+	auto Messages = View.FlushLocalChanges();
+
+	// THEN
+	TestTrue("WorkerView has multiple CreateEntityRequest", Messages->CreateEntityRequests.Num() > 1);
+
+	return true;
+}
+
+WORKERVIEW_TEST(GIVEN_WorkerView_with_one_op_enqued_WHEN_GenerateViewDelta_called_THEN_ViewDelta_with_one_op_returned)
+{
+	// GIVEN
+	WorkerView View;
+	// TODO(Alex): what type should it be
+	TArray<Worker_Op> Ops;
+	Ops.Push(CreateEmptyCreateEntityResponseOp());
+	auto OpList = MakeUnique<ViewDeltaLegacyOpList>(Ops);
+	View.EnqueueOpList(MoveTemp(OpList));
+
+	// WHEN
+	auto ViewDelta = View.GenerateViewDelta();
+
+	// THEN
+	// TODO(Alex): should it be using LegacyOpList?
+	TestTrue("ViewDelta has one op", ViewDelta->GenerateLegacyOpList()->GetCount() == 1);
+
+	return true;
+}
+
+WORKERVIEW_TEST(GIVEN_WorkerView_with_multiple_ops_engued_WHEN_GenerateViewDelta_called_THEN_ViewDelta_with_multiple_ops_returned)
```

Comment:
This name isn't generally true. I'd probably avoid writing any tests that involve generating an op list from the view delta at the moment. We haven't written most of the things that would make it viable and they are hard tests to get right. I'd probably leave these ones until we have completed more of the view delta if we do them at all.

Reply:
Or put in other worlds - this is testing something we want to remove and making these tests more useful than doing things by inspection is harder than testing most other things.

Going from view delta to op list is a relatively straight forward process, but testing that you have the right op list and that you have enough cases is hard. Testing the view delta is correct is much easier.